tests/test_sumpsf.py

```diff
@@ -0,0 +1,507 @@
+        print('comp0: ',test_star.fit.params[0],' =? ',true_params1)
+        np.testing.assert_almost_equal(test_star.fit.params[0], true_params1, decimal=2)
+        true_params2 = [ 0.2, 0, -0.7 + 1.5*y_list[i]/2048, 1.2 + 0.3*x_list[i]/2048, 0, 0 ]
+        print('comp1: ',test_star.fit.params[1],' =? ',true_params2)
+        np.testing.assert_almost_equal(test_star.fit.params[1], true_params2, decimal=2)
+
+        # The drawn image should be reasonably close to the target.
+        target = target[test_star.image.bounds]
+        print('target max diff = ',np.max(np.abs(test_star.image.array - target.array)))
+        np.testing.assert_allclose(test_star.image.array, target.array, atol=1.e-5)
+
+        # Draw should produce something almost identical (modulo bounds).
+        test_image = psf.draw(x=test_star['x'], y=test_star['y'],
+                              stamp_size=config['input']['stamp_size'],
+                              flux=test_star.fit.flux, offset=test_star.fit.center/image.scale)
+        b = test_star.image.bounds & test_image.bounds
+        print('image max diff = ',np.max(np.abs(test_image[b].array - test_star.image[b].array)))
+        np.testing.assert_allclose(test_image[b].array, test_star.image[b].array, atol=1.e-8)
+
+    # Repeat with both components being PixelGrid mdoels.
+    # Both pretty chunky, so this test doesn't take forever.
+    config['psf']['components'][0]['model'] = {
+        'type': 'PixelGrid',
+        'scale': 1.04,  # 4x4 native pixels per grid pixel.
+        'size': 10,     # Covers 40x40 original pixels.
+    }
+    if __name__ == '__main__':
+        # The small Gaussian moves around the central ~8 pixels.
+        # Make the grid big enough to capture the whole moving Gaussian.
+        grid_size = 14
+        tol = 3.e-4
+    else:
+        # For faster running on CI, use a bit smaller central piece with higher tolerance.
+        grid_size = 10
+        tol = 5.e-4
+    config['psf']['components'][1]['model'] = {
+        'type': 'PixelGrid',
+        'scale': 0.26,  # native pixel scale.
+        'size': grid_size,
+        'init': 'zero',
+        'centered': False,
+        'fit_flux': True,
+    }
+    psf = piff.process(config, logger)
+    assert type(psf) is piff.SumPSF
+    assert len(psf.components) == 2
+    assert type(psf.components[0]) is piff.SimplePSF
+    assert type(psf.components[0].model) is piff.PixelGrid
+    assert type(psf.components[0].interp) is piff.Mean
+    assert type(psf.components[1]) is piff.SimplePSF
+    assert type(psf.components[1].model) is piff.PixelGrid
+    assert type(psf.components[1].interp) is piff.Polynomial
+
+    for i, star in enumerate(psf.stars):
+        print('star',i)
+        target = targets[i]
+        test_star = psf.drawStar(star)
+        # Note: There is no useful test to make of the component parameters.
+        #print('comp0: ',test_star.fit.params[0])
+        #print('comp1: ',test_star.fit.params[1])
```

Comment:
What do you mean here?

Reply:
Usually I had been testing the fitted parameters to some known true values.  But for this test, both models are PixelGrid, so the parameters aren't anything I know a priori.  The only thing to test is to use the models to draw images of the total PSF model and see if it matches the data acceptably.